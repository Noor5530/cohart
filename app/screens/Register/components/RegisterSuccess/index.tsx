import React from "react";
import { View, Text, TouchableOpacity, Image } from "react-native";

import ArrowRight from "assets/Arrow_Right.png";
import styles from "./style";

interface IProps {
  handleShowSocial(): void;
  onNavigateToReferFriend: () => void;
}
const RegisterSuccess: React.FC<IProps> = ({ handleShowSocial, onNavigateToReferFriend }: IProps) => {
  return (
    <View style={styles.registerContainer}>
      <View
        style={styles.registerSubContainer}
      />
      <Text style={styles.registerSuccessHeading}>
        Congrats! Youâ€™ve been added to the list
      </Text>
      <Text style={[styles.descriptionText, styles.marginSubDescription]}>
        Please head to your email to opt-in. You can track your place on the
        website.
      </Text>
      <Text style={[styles.descriptionText, styles.subDescriptionText]}>
        Invite a friend to move one step closer in line :)
      </Text>
      <View style={[styles.buttonGroup, styles.registerSuccessButtonGroup]}>
        <TouchableOpacity
          onPress={handleShowSocial}
          style={[styles.buttonStyle]}
        >
          <Text style={[styles.buttonText, styles.buttonTextGray]}>
            Maybe Later
          </Text>
        </TouchableOpacity>
        <TouchableOpacity
          style={[styles.buttonStyle, styles.primaryButton]}
          onPress={onNavigateToReferFriend}
        >
          <Text
            style={[
              styles.buttonText,
              styles.buttonTextBlack,
              styles.buttonTextWithIcon,
            ]}
          >
            Let&apos;s do it
          </Text>
          <Image source={ArrowRight} />
        </TouchableOpacity>
      </View>
    </View>
  );
};

export default RegisterSuccess;
