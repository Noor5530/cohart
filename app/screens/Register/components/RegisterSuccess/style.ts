import { StyleSheet } from "react-native";
import {
  widthPercentageToDP as wp,
  heightPercentageToDP as hp,
} from "react-native-responsive-screen";

import { appFonts } from "components/text";
import { primary } from "../../../../config/colors";

const styles = StyleSheet.create({
  registerContainer: {
    flex: 1,
    backgroundColor: "white"
  },
  registerSubContainer: {
    width: 45,
    height: 7,
    backgroundColor: "rgba(0, 0, 0, 0.4)",
    borderRadius: 12,
    alignSelf: "center",
    marginTop: 10,
  },
  registerSuccessHeading: {
    fontFamily: appFonts.InterSemiBold,
    fontSize: 40,
    alignSelf: "center",
    marginTop: hp("7"),
    marginBottom: 40,
    width: wp("82"),
  },
  descriptionText: {
    width: wp("82"),
    fontFamily: appFonts.InterRegular,
    fontSize: 18,
    alignSelf: "center",
  },
  marginSubDescription: {
    marginBottom: hp("3"),
  },
  subDescriptionText: {
    marginBottom: hp("10"),
  },
  registerSuccessButtonGroup: {
    marginTop: hp("2"),
    paddingLeft: wp("6"),
    paddingRight: wp("6"),
  },
  buttonTextWithIcon: {
    marginRight: wp("2.2"),
  },
  buttonText: {
    textTransform: "uppercase",
    fontWeight: "bold",
  },
  buttonTextGray: {
    color: "gray",
  },
  buttonTextBlack: {
    color: "black",
  },
  buttonGroup: {
    flexDirection: "row",
    justifyContent: "space-around",
    marginTop: hp("2"),
  },
  buttonStyle: {
    height: 36,
    width: wp("40"),
    borderRadius: 24.01,
    justifyContent: "center",
    alignItems: "center",
    flexDirection: "row",
  },
  primaryButton: {
    backgroundColor: primary,
  },
});

export default styles;
