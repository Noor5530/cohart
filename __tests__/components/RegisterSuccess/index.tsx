import "react-native";
import React from "react";
import RegisterSuccess from "../../../app/screens/Register/components/RegisterSuccess";
import renderer from "react-test-renderer";

jest.useFakeTimers();

test("renders correctly", () => {
  const tree = renderer
    .create(<RegisterSuccess handleShowSocial={jest.fn} />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
