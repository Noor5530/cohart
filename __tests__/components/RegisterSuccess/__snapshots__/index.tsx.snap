// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "rgba(0, 0, 0, 0.4)",
        "borderRadius": 12,
        "height": 7,
        "marginTop": 10,
        "width": 45,
      }
    }
  />
  <Text
    style={
      Object {
        "alignSelf": "center",
        "fontFamily": "Inter-SemiBold",
        "fontSize": 40,
        "marginBottom": 40,
        "marginTop": undefined,
        "width": undefined,
      }
    }
  >
    Congrats! Youâ€™ve been added to the list
  </Text>
  <Text
    style={
      Array [
        Object {
          "alignSelf": "center",
          "fontFamily": "Inter-Regular",
          "fontSize": 18,
          "width": undefined,
        },
        Object {
          "marginBottom": undefined,
        },
      ]
    }
  >
    Please head to your email to opt-in. You can track your place on the website.
  </Text>
  <Text
    style={
      Array [
        Object {
          "alignSelf": "center",
          "fontFamily": "Inter-Regular",
          "fontSize": 18,
          "width": undefined,
        },
        Object {
          "marginBottom": undefined,
        },
      ]
    }
  >
    Invite a friend to move one step closer in line :)
  </Text>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "justifyContent": "space-around",
          "marginTop": undefined,
        },
        Object {
          "marginTop": undefined,
          "paddingLeft": undefined,
          "paddingRight": undefined,
        },
      ]
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderRadius": 24.01,
          "flexDirection": "row",
          "height": 36,
          "justifyContent": "center",
          "opacity": 1,
          "width": undefined,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontWeight": "bold",
              "textTransform": "uppercase",
            },
            Object {
              "color": "gray",
            },
          ]
        }
      >
        Maybe Later
      </Text>
    </View>
    <View
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#E6FF00",
          "borderRadius": 24.01,
          "flexDirection": "row",
          "height": 36,
          "justifyContent": "center",
          "opacity": 1,
          "width": undefined,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontWeight": "bold",
              "textTransform": "uppercase",
            },
            Object {
              "color": "black",
            },
            Object {
              "marginRight": undefined,
            },
          ]
        }
      >
        Let's do it
      </Text>
      <Image
        source={
          Object {
            "testUri": "../../../app/assets/Arrow_Right.png",
          }
        }
      />
    </View>
  </View>
</View>
`;
